src/pages/networks/forms/NetworkLocalPeeringForm.tsx

```diff
@@ -0,0 +1,220 @@
+      <Input
+        {...getFormProps("name")}
+        type="text"
+        label="Name"
+        required
+        autoFocus
+        disabled={isEditing || !!editRestriction}
+        title={editRestriction}
+      />
+      <AutoExpandingTextArea
+        {...getFormProps("description")}
+        label="Description"
+        disabled={!!editRestriction}
+        title={editRestriction}
+      />
+      {isEditing ? (
+        <Input
+          id="customTargetProject"
+          type="text"
+          label="Target project"
+          value={formik.values.targetProject}
+          disabled={isEditing}
+          title={"Target project cannot be changed"}
+          required
+        />
+      ) : (
+        <ProjectSelector
+          id="targetProject"
+          name="targetProject"
+          label="Target project"
+          value={formik.values.targetProject}
+          setValue={(value) => {
+            formik.setFieldValue("targetProject", value, false);
+            if (value !== projectOtherLabel) {
+              formik.setFieldValue("customTargetProject", "", false);
+              formik.setFieldValue("targetNetwork", "", false);
+              formik.validateForm();
+            }
+            if (value === projectOtherLabel) {
+              formik.setFieldValue("targetNetwork", networkOtherLabel, false);
+            }
+            formik.validateForm();
+          }}
+          disabled={!!editRestriction}
+          projects={projectList}
+          required
+        />
+      )}
+
+      {formik.values.targetProject === projectOtherLabel && (
+        <Input
+          id="customTargetProject"
+          name="customTargetProject"
+          type="text"
+          placeholder="Enter target project name"
+          value={formik.values.customTargetProject}
+          onChange={formik.handleChange}
+          onBlur={formik.handleBlur}
+          disabled={isEditing || !!editRestriction}
+          title={editRestriction}
+        />
+      )}
+
+      {isEditing ? (
+        <Input
+          id="customTargetNetwork"
+          type="text"
+          label="Target network"
+          value={formik.values.targetNetwork}
+          disabled={isEditing}
+          title={"Target network cannot be changed"}
+          required
+        />
+      ) : (
+        formik.values.targetProject &&
+        formik.values.targetProject !== projectOtherLabel && (
+          <NetworkSelector
+            id="targetNetwork"
+            name="targetNetwork"
+            label="Target network"
+            value={formik.values.targetNetwork}
+            setValue={(value) => {
+              formik.setFieldValue("targetNetwork", value, false);
+              if (value !== networkOtherLabel) {
+                formik.setFieldValue("customTargetNetwork", "", false);
+              }
+              formik.validateForm();
+            }}
+            disabled={isEditing || !!editRestriction}
+            managedNetworks={networkList}
```

Comment:
```suggestion
            networkList={networkList}
```
I think the prop on the `NetworkSelector` has a bit of an odd name. Shall we call it `networkList`? Because the selector has no opinion of which networks it displays and if they are managed or not.